### PR TITLE
Various appleseed-Maya fixes

### PR DESCRIPTION
**Pull request by Mango-3**
- Add new build options header to source files
- Add SSE4.2 compiler directive for Linux
- Tweak adaptive sampler defaults

### CODE REVIEW

File: scripts/appleseedMaya/renderGlobals.py [hunk 1]
```diff
@@ -483,7 +483,7 @@ def create(self):
                             ui=pm.intSliderGrp(
                                 label="Max Samples",
                                 field=True,
-                                value=256,
+                                value=128,
```
Comment (dictoon) [inline scripts/appleseedMaya/renderGlobals.py:486]: It's 256 in appleseed.studio, and IIRC a recent commit of yours aligned appleseed-maya with appleseed.studio, so why this change?

File: scripts/appleseedMaya/renderGlobals.py [hunk 2]
```diff
@@ -518,7 +518,7 @@ def create(self):
                                 minValue=0.0001,
                                 fieldMinValue=0.0,
                                 maxValue=2.0,
-                                fieldMaxValue=10000.0,
+                                fieldMaxValue=25.0,
```
Comment (dictoon) [inline scripts/appleseedMaya/renderGlobals.py:521]: Same question here.

File: src/appleseedmaya/exporters/dagnodeexporter.h [hunk 3]
```diff
@@ -29,13 +29,18 @@
 #ifndef APPLESEED_MAYA_EXPORTERS_DAGNODEEXPORTER_H
 #define APPLESEED_MAYA_EXPORTERS_DAGNODEEXPORTER_H
 
+#pragma once
```
Comment (dictoon) [inline src/appleseedmaya/exporters/dagnodeexporter.h]: If you introduce `#pragma once` (which is a great move) then you need to remove the header guards.

**Review by dictoon — APPROVED** [review]

File: scripts/appleseedMaya/renderGlobals.py (hunk 1 above)
Comment (Mango-3) [reply scripts/appleseedMaya/renderGlobals.py:486]: In Maya the default Max value serves also as the default sample value for uniform sampler. If one uses the uniform sampler default 256 samples is a bit too high. In studio default for uniform is 64, so 128 is a compromise to not have too low default Max samples when the adaptive sampler is chosen.
I guess @est77 preferred the simpler UI-layout without the need to add an additional control for uniform samples.

File: src/appleseedmaya/exporters/dagnodeexporter.h (hunk 3 above)
Comment (Mango-3) [reply src/appleseedmaya/exporters/dagnodeexporter.h]: So you prefer to remove the header guards? I put #pragma once there during bug hunting and forgot to take it out.

File: scripts/appleseedMaya/renderGlobals.py (hunk 2 above)
Comment (Mango-3) [reply scripts/appleseedMaya/renderGlobals.py:521]: It is just limited to 25 in the UI, which is very high already. It can be overridden with MEL or Python if one would need to up to 10000, which is the hard limit set in the renderGlobals.cpp

File: scripts/appleseedMaya/renderGlobals.py (hunk 1 above)
Comment (dictoon) [reply scripts/appleseedMaya/renderGlobals.py:486]: Ah ok, makes sense then.

File: scripts/appleseedMaya/renderGlobals.py (hunk 2 above)
Comment (dictoon) [reply scripts/appleseedMaya/renderGlobals.py:521]: Ok.

File: src/appleseedmaya/exporters/dagnodeexporter.h (hunk 3 above)
Comment (dictoon) [reply src/appleseedmaya/exporters/dagnodeexporter.h]: We fully transitioned appleseed core from header guards to `#pragma once` since they are simpler, less error prone, possibly also faster and universally supported. At some point it would be nice to do the same in appleseed-maya, but not necessarily in this PR.

Comment (Mango-3) [reply src/appleseedmaya/exporters/dagnodeexporter.h]: Agree, I removed the #pragma once
This is better done in a separate PR and not mix more together. 
This PR is already a mix of unrelated issues.

Comment (dictoon) [reply src/appleseedmaya/exporters/dagnodeexporter.h]: Agreed, perfect.

**Review by dictoon — APPROVED** [review]
Looks good! Could you fuse commits 2 and 4 before merging please?

**Review by dictoon — APPROVED** [review]
